### PR TITLE
Split up hooks into separate modules

### PR DESCRIPTION
**Pull request by smarlowucf**
To distinguish the hookspec categories:

- csp_hookspecs for csp and billing interaction
- storage_hookspecs for CRUD of cache and CSP config
- hookspecs for everything else

### CODE REVIEW

**Review by rtamalin — APPROVED** [review]
LGTM

**Review by KeithMnemonic — APPROVED** [review]
LGTM

File: csp_billing_adapter/csp_hookspecs.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+def meter_billing(
+    config: Config,
+    dimensions: dict,
+    timestamp: str,
+):
+    """
+    Process metering request against the CSP API.
+
+    :param config: The application configuration dictionary
+    :param dimensions: A hash of the billing dimensions and quantities
```
Comment (rjschwei) [inline csp_billing_adapter/csp_hookspecs.py:34]: Aren't the dimensions part of the adapter config?

Comment (smarlowucf) [reply csp_billing_adapter/csp_hookspecs.py:34]: Yep. And there's a separate mapping of dimensions to usage for metering a bill.

**Pull request merged**